R/randomForest_tidiers.R

```diff
@@ -916,24 +975,46 @@ augment.rpart.classification <- function(x, data = NULL, newdata = NULL, ...) {
     newdata %>% dplyr::rename(predicted_label = predicted_value_col) %>%
                 dplyr::select(-predicted_label, everything(), predicted_label)
   } else if (!is.null(data)) {
+    if (nrow(data) == 0) {
+      # Handle the case where, for example, test_rate is 0 here,
+      # rather than trying to make it pass through following code, which can be complex.
+      return (data)
+    }
     y_value <- attributes(x)$ylevels[x$y]
     predicted_value_col <- avoid_conflict(colnames(data), "predicted_value")
-    predicted_probability_col <- avoid_conflict(colnames(newdata), "predicted_probability")
-    predicted_value <- x$predicted_class
-    predicted_probability <- apply(predict(x, data, type="prob"), 1, max)
+    predicted_probability_col <- avoid_conflict(colnames(data), "predicted_probability")
+    switch(data_type,
+      training = {
+        predicted_value_nona <- x$predicted_class
+        # TODO: This restore_na seems to be needed here, but it seems rpart is returning result with NA rows included for regression case. Whhat is making the difference?
```

Comment:
Get to the bottom of it.

Reply:
This was because, for regression, we were calling predict() with training data fed as new data. Changed not to feed training data again to predict(), which would then just return the prediction result it keeps internally, for both consistency and performance.